src/main/java/com/spotify/docker/client/DefaultDockerClient.java

```diff
@@ -414,7 +415,11 @@ protected DefaultDockerClient(final Builder builder) {
         .property(ApacheClientProperties.CONNECTION_MANAGER, cm)
         .property(ApacheClientProperties.REQUEST_CONFIG, requestConfig);
 
-    this.registryAuth = builder.registryAuth;
+    this.registryAuthSupplier = builder.registryAuthSupplier;
+
+    if (this.registryAuthSupplier == null) {
+      this.registryAuthSupplier = new NoOpRegistryAuthSupplier(null);
```

Comment:
would be better to have a no-arg constructor to NoOpRegistryAuthSupplier that can internally set the field to null, rather than the caller